compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1805,13 +1805,14 @@ fn add_local_native_libraries(
     let search_path = archive_search_paths(sess);
     let mut last = (NativeLibKind::Unspecified, None);
     for lib in relevant_libs {
-        // Skip if this library is the same as the last.
-        last = if (lib.kind, lib.name) == last { continue } else { (lib.kind, lib.name) };
-
         let name = match lib.name {
             Some(l) => l,
             None => continue,
         };
+
+        // Skip if this library is the same as the last.
+        last = if (lib.kind, lib.name) == last { continue } else { (lib.kind, lib.name) };
```

Comment:
This change is not strictly necessary for the fix, but let's leave it for consistency.
It also improves deduplication in the (unlikely) case of `#[link(name = "foo")] #[link(wasm_import_module = "...")] #[link(name = "foo")]`.